sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java

```diff
@@ -919,6 +941,15 @@ private RunQueryResponse runQueryWithRetries(RunQueryRequest request) throws Exc
         }
       }
 
+      protected GoogleJsonError.ErrorInfo getErrorInfo(Exception e) {
+        if (!(e instanceof GoogleJsonResponseException)) {
+          return null;
+        }
+        GoogleJsonError jsonError = ((GoogleJsonResponseException) e).getDetails();
+        GoogleJsonError.ErrorInfo errorInfo = Iterables.getFirst(jsonError.getErrors(), null);
+        return errorInfo;
+      }
+
```

Comment:
This PR only instruments reading from Datastore, but not writing.

Please instrument here as well. (You can do in a separate PR if you like, if so rename this PR to 
"[BEAM-11983] Java Datastore - Implement IO Request Count metrics for Datastore Reads"

https://github.com/apache/beam/blob/fc24debff70dba842757dc249a7362abbdb34c75/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java#L1341

Reply:
Added metrics to write